app/src/main/java/net/gsantner/markor/ui/SearchOrCustomTextDialogCreator.java

```diff
@@ -391,6 +395,7 @@ public static void showSearchDialog(Activity activity, Editable edit, int[] sel,
     public static void showTodoSearchDialog(Activity activity, Editable edit, int[] sel, Callback.a1<Spannable> highlighter, Callback.a1<List<Integer>> userCallback) {
         SearchOrCustomTextDialog.DialogOptions dopt2 = basicSearchDialogopts(activity, edit, sel);
         dopt2.isMultiSelectEnabled = true;
+        dopt2.neutralButtonText = R.string.deselect;
```

Comment:
@gsantner 

Now the search / replace option isn't present

With the old approach you got `Deselect` when multiple options were selected and `Search / replace` when none were selected.

Reply:
Where do we need this? Search/Replace is only important for the document top-toolbar search button, which not has Multi-Selection anyway

Reply:
This change removed search / replace for all file types.

> which not has Multi-Selection anyway

I had multi selection enabled for the top search for Todo file type, where it is useful.

I'll make a PR to fix this tomorrow

Reply:
The other option is to keep the neutral button constant and turn _cancel_ into Deselect when more than 1 item selected.

Do you prefer this?

Reply:
Humble said I generally don't see much of the need for "Unselect all". Especially not for todo, think it might hurt more than it brings [oh no, need to recheck all my finished stuff].

So if you goal to improve for todo.txt I rather suggest to keep the neutral slot free for an additional action.

> The other option is to keep the neutral button constant and turn cancel into Deselect when more than 1 item selected.

No, at dialog it's important to have a button to go out without changes saved. We should not add a dual-use for it

Reply:
An important point also, in case you think off synchronizing the checked states with editor: You can only set cursor across one rectangle of items, if there is any non-selected in between, it will be spanned by the selection as well.

Reply:
> plan to do with the checkboxes atm - i.e. apply the "done" states

Apply done, change priority, add context or project, add or change due date. I do all of these for multiple tasks all the time.

> I generally don't see much of the need for "Unselect all"

Ok. I guess the user could always cancel and come back if they needed to. What we can do is just add the selection count in parenthesis after the _title_? This will also simplify the logic a great deal.

Reply:
So what did you have in mind with "Unselect all", do you really use or plan to use it often? Or just a nice to have?

Not wanna destroy your use case. Just guess most of the time it won't be used

>  What we can do is just add the selection count in parenthesis after the title? This will also simplify the logic a great deal.

It's a option, in case we even want to display the amount of selected items. Could be Title [x] or also at OK [x] button

Reply:
It seemed natural to have an unselect all. I don't seem to actually use it that often so. We can add it back if I find that I or others really feel the need for it.

> or also at OK [x] button. 

I like this more than title (x) actually. Makes it clearer.